src/formats/notion-api/database-helpers.ts

```diff
@@ -0,0 +1,864 @@
+		return `<!-- Failed to import database: ${error.message} -->`;
+	}
+}
+
+/**
+ * Query all pages from a database with pagination support
+ * Note: In Notion API v2025-09-03, databases are now called "data sources"
+ */
+async function queryAllDatabasePages(
+	client: Client,
+	databaseId: string,
+	ctx: ImportContext
+): Promise<PageObjectResponse[]> {
+	const pages: PageObjectResponse[] = [];
+	let cursor: string | undefined = undefined;
+	
+	do {
+		// In Notion API v2025-09-03, use dataSources.query instead of databases.query
+		const response: any = await makeNotionRequest(
+			() => client.dataSources.query({
+				data_source_id: databaseId,
+				start_cursor: cursor,
+				page_size: 100,
+			}),
+			ctx
+		);
+		
+		// Filter to get full page objects
+		const fullPages = response.results.filter(
+			(page: any): page is PageObjectResponse => page.object === 'page'
+		);
+		
+		pages.push(...fullPages);
+		cursor = response.has_more ? response.next_cursor ?? undefined : undefined;
+		
+	} while (cursor);
+	
+	return pages;
+}
+
+/**
+ * Extract database title from Notion database object
+ */
+function extractDatabaseTitle(database: DatabaseObjectResponse): string {
+	if (database.title && database.title.length > 0) {
+		return database.title.map(t => t.plain_text).join('');
+	}
+	return 'Untitled Database';
+}
+
+/**
+ * Create a .base file for the database
+ */
+export async function createBaseFile(
+	vault: Vault,
+	databaseName: string,
+	databaseFolderPath: string,
+	outputRootPath: string,
+	properties: any,
+	databasePages: PageObjectResponse[],
+	formulaStrategy: FormulaImportStrategy = 'function',
+	propertyIds: string[] = []
+): Promise<string> {
+	// Create "Notion Databases" folder at the same level as output root
+	const parentPath = outputRootPath.split('/').slice(0, -1).join('/') || '/';
+	const databasesFolder = parentPath === '/' ? '/Notion Databases' : parentPath + '/Notion Databases';
+	
+	try {
+		let folder = vault.getAbstractFileByPath(normalizePath(databasesFolder));
+		if (!folder) {
+			await vault.createFolder(normalizePath(databasesFolder));
+		}
+	}
+	catch (error) {
+		console.error('Failed to create Notion Databases folder:', error);
+	}
+	
+	// Generate .base file content
+	const baseContent = generateBaseFileContent(databaseName, databaseFolderPath, properties, databasePages, formulaStrategy, propertyIds);
+	
+	// Create .base file
+	const baseFilePath = normalizePath(`${databasesFolder}/${databaseName}.base`);
+	
+	// Check if file already exists, if so, add number suffix
+	let finalPath = baseFilePath;
+	let counter = 1;
+	while (vault.getAbstractFileByPath(finalPath)) {
+		finalPath = normalizePath(`${databasesFolder}/${databaseName} (${counter}).base`);
+		counter++;
+	}
+	
+	await vault.create(finalPath, baseContent);
+	
+	return finalPath;
+}
+
+/**
+ * Generate content for .base file
+ */
+function generateBaseFileContent(
+	databaseName: string,
+	databaseFolderPath: string,
+	properties: any,
```

Comment:
Make this name consistent through all the functions the value is passed through. `notionProperties` or `datasourceProperties` would be better than `properties`.